File: ShiftSync.API/Models/Entities/BreakType.cs
namespace ShiftSync.API.Models.Entities;

public class BreakType
{
    public int Id { get; set; }

    public string Name { get; set; } = string.Empty;

    public int MaxDurationMinutes { get; set; }
    public int MaxOccurrencesPerShift { get; set; } = 1;
    public bool IsActive { get; set; } = true;

    public ICollection<AttendanceBreak> AttendanceBreaks { get; set; } = new List<AttendanceBreak>();
}

File: ShiftSync.Web/src/types/shiftsync/breakType.ts
export interface BreakType {
  id: number;
  name: string;
  maxDurationMinutes: number;
  maxOccurrencesPerShift: number;
  isActive: boolean;
}

export interface CreateBreakTypeDto {
  name: string;
  maxDurationMinutes: number;
  maxOccurrencesPerShift: number;
}

export interface UpdateBreakTypeDto {
  name?: string;
  maxDurationMinutes?: number;
  maxOccurrencesPerShift?: number;
}

export interface BreakTypeFilter {
  pageNumber?: number;
  pageSize?: number;
  search?: string;
  isActive?: boolean;
}
